src/ngraph_backend_manager.cc

```diff
@@ -109,5 +112,47 @@ bool BackendManager::IsSupportedBackend(const string& backend_name) {
   return true;
 };
 
+// Backend Config functions
+BackendConfig* BackendManager::GetBackendConfig(const string& backend_name) {
+  std::lock_guard<std::mutex> lock(BackendManager::ng_backend_map_mutex_);
+  auto itr = BackendManager::ng_backendconfig_map_.find(backend_name);
+  if (itr == BackendManager::ng_backendconfig_map_.end()) {
+    BackendConfig* bconfig = nullptr;
+    if (backend_name == "NNPI") {
+      NGRAPH_VLOG(3) << "Creating NNPI Backend config";
+      bconfig = new BackendNNPIConfig();
+    } else {
+      NGRAPH_VLOG(3) << "Creating default Backend config";
+      bconfig = new BackendConfig(backend_name);
+    }
+    BackendManager::ng_backendconfig_map_[backend_name] = bconfig;
```

Comment:
We may want to free `bconfig` in `~BackendManager` (by freeing all the values of ng_backendconfig_map_) or save it as smart pointers for auto-freeing